plugin-fancymenu/lxqtfancymenu.cpp

```diff
@@ -0,0 +1,388 @@
+            return;
+        }
+    }
+
+    loadFavorites();
+    setMenuFontSize();
+
+    //clear the search to not leaving the menu in wrong state
+    mFilterClear = settings()->value(QStringLiteral("filterClear"), false).toBool();
+    mWindow->setFilterClear(mFilterClear);
+
+    bool buttonsAtTop = settings()->value(QStringLiteral("buttonsAtTop"), false).toBool();
+    mWindow->setButtonPosition(buttonsAtTop ? LXQtFancyMenuButtonPosition::Top : LXQtFancyMenuButtonPosition::Bottom);
+
+    bool categoriesAtRight = settings()->value(QStringLiteral("categoriesAtRight"), true).toBool();
+    mWindow->setCategoryPosition(categoriesAtRight ? LXQtFancyMenuCategoryPosition::Right : LXQtFancyMenuCategoryPosition::Left);
+
+    realign();
+}
+
+/************************************************
+
+ ************************************************/
+void LXQtFancyMenu::buildMenu()
+{
+    mWindow->rebuildMenu(mXdgMenu);
+
+    mWindow->doSearch();
+    setMenuFontSize();
+}
+
+void LXQtFancyMenu::loadFavorites()
+{
+    bool listChanged = false;
+
+    const QList<QMap<QString, QVariant> > list = settings()->readArray(QStringLiteral("favorites"));
+    QStringList fileList;
+    for(const QMap<QString, QVariant>& item : list)
+    {
+        QString file = item.value(QStringLiteral("desktopFile")).toString();
+        if(file.isEmpty())
+        {
+            listChanged = true;
+            continue;
+        }
+
+        QString canonicalPath = QDir(file).canonicalPath();
+        if(canonicalPath != file)
+            listChanged = true;
+
+        if(canonicalPath.isEmpty())
+            continue;
+
+        if(fileList.contains(canonicalPath))
+        {
+            // Don't add duplicates
+            listChanged = true;
+            continue;
+        }
+
+        fileList.append(canonicalPath);
+    }
+
+    mWindow->setFavorites(fileList);
+
+    if(listChanged)
+        saveFavorites();
+}
+
+void LXQtFancyMenu::saveFavorites()
+{
+    const QStringList fileList = mWindow->favorites();
+
+    QList<QMap<QString, QVariant> > list;
+    for(const QString& file : fileList)
+    {
+        QMap<QString, QVariant> item;
+        item.insert(QStringLiteral("desktopFile"), file);
+        list.append(item);
+    }
+
+    settings()->setArray(QStringLiteral("favorites"), list);
+}
+
+/************************************************
+
+ ************************************************/
+void LXQtFancyMenu::setMenuFontSize()
+{
+    if (!mWindow)
+        return;
+
+    QFont menuFont = mButton.font();
+    bool customFont = settings()->value(QStringLiteral("customFont"), false).toBool();
+
+    if(customFont)
+    {
+        menuFont = mWindow->font();
+        menuFont.setPointSize(settings()->value(QStringLiteral("customFontSize")).toInt());
+    }
+
+    if (mWindow->font() != menuFont)
+    {
+        mWindow->setFont(menuFont);
+    }
+
+    // FIXME: font is not really changing, what about icon sizes?
```

Comment:
To really change the font, you could do this:

The condition `mWindow->font() != menuFont` doesn't do anything because the font itself isn't changed. Add a public method `void setCustomFont(const QFont &font);` to the `LXQtFancyMenuWindow` class as

```c++
void LXQtFancyMenuWindow::setCustomFont(const QFont &font)
{
    mAppView->setFont(font);
    mCategoryView->setFont(font);
}
```

Then use this inside `LXQtFancyMenu::setMenuFontSize()`:

```c++
    if(customFont)
    {
        menuFont = mWindow->font();
        menuFont.setPointSize(settings()->value(QStringLiteral("customFontSize")).toInt());
        mWindow->setCustomFont(menuFont);
    }
```

Reply:
From [Qt docs](https://doc.qt.io/qt-5/qwidget.html#font-prop) I understood it would propagate font changes to child widgets but it's not working.
Adding `setCustomFont()` it works.

Reply:
> From [Qt docs](https://doc.qt.io/qt-5/qwidget.html#font-prop) I understood...

Yes. Qt's code sometimes contradicts Qt's doc.